lib/sendgrid-actionmailer.rb

```diff
@@ -17,10 +17,13 @@ def initialize(params)
     end
 
     def deliver!(mail)
+      from = mail[:from].tree.addresses.first
```

Comment:
Yeah, From is arbitrary and really we mean Sender, which must be singular, but this is life.